eZ/Publish/API/Repository/Tests/SearchServiceFulltextEmbedTest.php

```diff
@@ -0,0 +1,181 @@
+            $this->markTestSkipped('Engine says it does not support advance fulltext format');
+        }
+    }
+
+    public function testPrepareContent(): void
+    {
+        $contentService = $this->getRepository()->getContentService();
+        $baseArticleStruct = $this->prepareBaseArticleStruct();
+
+        $slaveArticleStruct = $this->fillSlaveArticleStruct(clone $baseArticleStruct);
+        $slaveArticleContent = $contentService->publishVersion(
+            $this->createContent($slaveArticleStruct)->versionInfo
+        );
+
+        $mainArticleStruct = $this->fillMainArticleStruct(clone $baseArticleStruct, $slaveArticleContent->id);
+        $mainArticleContent = $contentService->publishVersion(
+            $this->createContent($mainArticleStruct)->versionInfo
+        );
+
+        $this->refreshSearch($this->getRepository());
+
+        self::$createdIds = [
+            $slaveArticleContent->id,
+            $mainArticleContent->id,
+        ];
+    }
+
+    /**
+     * @depends testPrepareContent
+     */
+    public function testFulltextContentSearch(): void
+    {
+        $searchService = $this->getRepository()->getSearchService();
+
+        $query = new Query([
+            'query' => new Criterion\FullText(self::SLAVE_ARTICLE_NAME),
+        ]);
+
+        $searchResult = $searchService->findContent($query);
+
+        $this->assertGreaterThanOrEqual(2, $searchResult->totalCount);
+        $this->assertResults($searchResult->searchHits);
+    }
+
+    /**
+     * @depends testPrepareContent
+     */
+    public function testFulltextLocationSearch(): void
+    {
+        $searchService = $this->getRepository()->getSearchService();
+
+        $query = new LocationQuery([
+            'query' => new Criterion\FullText(self::SLAVE_ARTICLE_NAME),
+        ]);
+
+        $searchResult = $searchService->findLocations($query);
+
+        $this->assertGreaterThanOrEqual(2, $searchResult->totalCount);
+        $this->assertResults($searchResult->searchHits);
+    }
+
+    private function prepareBaseArticleStruct(): ContentCreateStruct
+    {
+        $introDocument = new \DOMDocument();
+        $introDocument->loadXML(
+            <<<EOT
+<?xml version="1.0" encoding="UTF-8"?>
+<section xmlns="http://docbook.org/ns/docbook" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:ezxhtml="http://ez.no/xmlns/ezpublish/docbook/xhtml" xmlns:ezcustom="http://ez.no/xmlns/ezpublish/docbook/custom" version="5.0-variant ezpublish-1.0">
+<para>some paragraph</para>
+</section>
+EOT
+        );
+
+        $repository = $this->getRepository();
+        $contentType = $repository->getContentTypeService()->loadContentTypeByIdentifier('article');
+
+        /** @var ContentCreateStruct $articleStruct */
+        $articleStruct = $repository->getContentService()->newContentCreateStruct($contentType, 'eng-GB');
+        $articleStruct->setField('intro', new RichTextValue($introDocument), 'eng-GB');
```

Comment:
Note: It will be difficult merge-up. RichText is gone from Kernel in `master (8.0@dev)` and so are the integration tests for it. I think we should start to move tests covering Solr capabilities to Solr Bundle.

Reply:
> It will be difficult merge-up. RichText is gone from Kernel in master (8.0@dev) and so are the integration tests for it.

I agree. AFAIK there is no other FT witch expose `\eZ\Publish\API\Repository\Values\Content\Relation::EMBED` relation which is essential to this test, so probably we should introduce "Embed" FT for integration test needs?

> I think we should start to move tests covering Solr capabilities to Solr Bundle.

Yes, but it should be done as a separate story. EZP-31226 is a bugfix (CR).